### PR TITLE
Updating guidelines for referring to API object terms

### PR DESCRIPTION
**Pull request by bergerhoffer**

### CODE REVIEW

File: contributing_to_docs/doc_guidelines.adoc [hunk 1]
```diff
@@ -863,21 +863,71 @@ Text for admonition
...
+.Examples of general references
+....
+Kubernetes runs your workload by placing containers into pods to run on nodes.
+
+You must have at least one secret, config map, or service account.
+
+The total number of persistent volume claims in a project.
+....
+
+Note that if an object uses an acronym or other special capitalization, then its general reference should honor that. For example, general references to `APIService` should be written as "API service", not "api service". Any other exceptions or special guidance are noted in the xref:../../contributing_to_docs/term_glossary.adoc[glossary].
```
Comment (lbarbeevargas) [inline contributing_to_docs/doc_guidelines.adoc:893]: Is `APIService` supposed to be `API Service` here?

File: contributing_to_docs/doc_guidelines.adoc [hunk 2]
```diff
@@ -863,21 +863,71 @@ Text for admonition
...
+When referring to API objects in general usage, use lowercase and separate multi-word API objects. By general reference, this means any time you are speaking conceptually, or generally, about these components in a cluster. *Default to following this guidance unless you are specifically interacting with/referring to the API object.*
+
+For example:
+
+* pod
+* node
+* daemon set
+* config map
+* deployment
+* deployment config
```
Comment (jboxman) [inline contributing_to_docs/doc_guidelines.adoc:880]: This is special, because someone might refer to a `Deployment` object configuration as simply a deployment configuration, as a general term. But that collides with _deployment config_, which is the lowercase version of the `DeploymentConfig` object, which is very different from the `Deployment` object. (I'm not sure _deployment config configuration_ makes sense, either, though.)

Because of this confusion, the deployment sections of the docs need somewhat of a refresh. (OCP also encourages use of `Deployment`, with `DeploymentConfig` eventually be deprecated someday.)

File: contributing_to_docs/doc_guidelines.adoc (hunk 1 above)
Comment (bergerhoffer) [reply contributing_to_docs/doc_guidelines.adoc:893]: @lbarbeevargas No, `APIService` is the correct API object kind. If something is monospace and PascalCase, then there should never be a space in there.

You can see the of this particular example in the spreadsheet, line 22: https://docs.google.com/spreadsheets/d/1VWJwd7Yk91CY_BnyCKSpHcPwuRNEYASmxkXRIRp4WEY/edit#gid=0

File: contributing_to_docs/doc_guidelines.adoc (hunk 2 above)
Comment (bergerhoffer) [reply contributing_to_docs/doc_guidelines.adoc:880]: @jboxman When I was updating the glossary entries for deployment and deployment config, I was trying to think if we should provide guidance on being more specific in these cases, so as to make the distinction clearer. But I couldn't think of what to say.

I'd be interested to hear if @adellape has any suggestions.

One thing that we could at least do is advise against saying "deployment configuration" to mean a `Deployment` object's configuration.

But it might help to take a look at the sections that we have on deployments and deployment configs, to see how it's going to look and see what guidance we need to provide.

Comment (jboxman) [reply contributing_to_docs/doc_guidelines.adoc:880]: I'd say in this case _deployment_ is the concept; And in OCP we happen to, unsurprisingly, have two different ways to accomplish that goal. So we probably need to talk about a deployment generally, and then what each specific `Deployment` or `DeploymentConfig` gets you as an OCP user.

We don't need to belabor `DeploymentConfig`, as from what I've read recently, we'd prefer everyone use upstream `Deployment`. It now supports more or less the functionality of `DeploymentConfig`.

Comment (bergerhoffer) [reply contributing_to_docs/doc_guidelines.adoc:880]: That sounds fair to me, though I was wondering if we'd still need to speak of deployment configs generally. But in doing a quick search in the repo, I see a few places where we already are calling them deployments, even though the CLI command is getting a deployment config.

We might need to try it out on a few modules to see how this plays out before we can decide on the guidance.

Comment (bergerhoffer) [reply contributing_to_docs/doc_guidelines.adoc:880]: @openshift/team-documentation In case anyone else has any thoughts on this ^

File: contributing_to_docs/term_glossary.adoc [hunk 3]
```diff
@@ -18,6 +18,8 @@ goal is to standardize terminology across OpenShift content and be consistent in
 the usage of our terminology when referring to OpenShift components or
 architecture.
 
+For terms that are also API objects, there is different guidance for general usage of the term versus referencing the actual API object. This glossary mainly defines the general usage guideline (lowercase, separating words), but be sure to use the object formatting (PascalCase, in monospace) when referring to the actual object. See link:doc_guidelines.adoc#api-object-formatting[API Object Formatting] for more information
```
Comment (bobfuru) [inline contributing_to_docs/term_glossary.adoc:21]: missing a period

File: contributing_to_docs/term_glossary.adoc [hunk 4]
```diff
@@ -391,75 +423,72 @@ menu that does not have hover text or a given name or label in the web console.
 == P
```
Comment (bobfuru) [inline contributing_to_docs/term_glossary.adoc:423]: Should we add a note for persistent volume and persistent volume claim similar to custom resource definition (CRD)?

> Usage: persistent volume and persistent volume claims
> 
> These are commonly abbreviated as a PV and a PVC.
> 
> Developers can use a PVC to request a PV resource without having specific knowledge of the underlying storage infrastructure.

File: contributing_to_docs/doc_guidelines.adoc (hunk 2 above)
Comment (adellape) [reply contributing_to_docs/doc_guidelines.adoc:880]: +1 to just sticking with to lowercase "deployments" for general references, and use `Deployments` and `DeploymentConfigs` when you need to be specific or distinguish. Also would be interested if/when we can actually sunset DCs from our docs.

File: contributing_to_docs/doc_guidelines.adoc [hunk 5]
```diff
@@ -863,21 +863,71 @@ Text for admonition
...
-This matches general Kubernetes usage and makes it obvious that a specific
-concept is being referred to. For example:
+[id="api-object-general-references"]
+=== General references
 
-- Pod
-- Deployment
-- Operator
-- DaemonSet (not "daemon set", "daemonset", "daemonSet", or "Daemonset")
+When referring to API objects in general usage, use lowercase and separate multi-word API objects. By general reference, this means any time you are speaking conceptually, or generally, about these components in a cluster. *Default to following this guidance unless you are specifically interacting with/referring to the API object.*
```
Comment (adellape) [inline contributing_to_docs/doc_guidelines.adoc:871]: The "By general reference..." sentence was a little hard to parse at first.

File: contributing_to_docs/term_glossary.adoc [hunk 6]
```diff
@@ -527,20 +554,22 @@ In addition to "spec file" being allowed related to RPM spec files, general
 usage of "spec" is allowed when describing Kubernetes or OpenShift object specs
 / manifests / definitions.
 
-*Examples of correct usage*
+.Examples of correct usage
 ====
-Update the Pod spec to reflect the changes.
+Update the `Pod` spec to reflect the changes.
```
Comment (adellape) [inline contributing_to_docs/term_glossary.adoc:559]: Is it worth adding "spec" up in the example list of alternatives to "object", "resource", etc? Don't need to list every single option, but since it's mentioned down below here...

File: contributing_to_docs/term_glossary.adoc (hunk 4 above)
Comment (adellape) [reply contributing_to_docs/term_glossary.adoc:423]: Maybe a generic entry for "volume" too?

File: contributing_to_docs/doc_guidelines.adoc (hunk 2 above)
Comment (bergerhoffer) [reply contributing_to_docs/doc_guidelines.adoc:880]: @jboxman @adellape I updated the glossary to use `Deployment` and `DeploymentConfig`. Can you take a look and see if that makes sense?

File: contributing_to_docs/term_glossary.adoc (hunk 4 above)
Comment (bergerhoffer) [reply contributing_to_docs/term_glossary.adoc:431]: Thanks @bobfuru I added entries for PV and PVC.

@adellape if you have a good description to use for just "volume", I'd be happy to add. Otherwise we can always add in a separate PR.

**Review by adellape — APPROVED** [review]
:fire:

**Comment by bergerhoffer** [comment]
Going to merge these guidelines in so that we can start following them. We can always add more entries to the glossary as needed for terms that need special handling.
